plugins/orchestrator-backend/src/service/router.ts

```diff
@@ -271,6 +279,11 @@ function setupInternalRoutes(
       params: { workflowId },
     } = req;
 
+    const workflowDefinition =
+      await dataIndexService.getWorkflowDefinition(workflowId);
+    const serviceUrl = workflowDefinition.serviceUrl ?? '';
```

Comment:
```suggestion
    const serviceUrl = workflowDefinition.serviceUrl;
    if (!serviceUrl) {
      throw new Error(`ServiceUrl is not defined for workflow ${workflowId}`);
    }
```
Similar to what you did in some lines above.